File: T2-Java/T4/codigos2223/Pestanias/src/main/java/com/pestanias/pestanias/model/Usuario.java
package com.pestanias.pestanias.model;

public class Usuario {

    private int id;
    private String nombre, apellido, correo;

    public Usuario(int id, String nombre, String apellido, String correo) {
        this.id = id;
        this.nombre = nombre;
        this.apellido = apellido;
        this.correo = correo;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellido() {
        return apellido;
    }

    public void setApellido(String apellido) {
        this.apellido = apellido;
    }

    public String getCorreo() {
        return correo;
    }

    public void setCorreo(String correo) {
        this.correo = correo;
    }

    @Override
    public String toString() {
        return getId() + " - " +getNombre();
    }
}

File: Tema5/codigos_ces/inicial/src/app/model/usuario.ts
export class usuario {
  // attri
  // const
  hobbies: string[];
  constructor(
    private nombre: string,
    private apellido: string,
    private edad: number,
  ) {
    this.hobbies = [];
  }
  // metodos -> para poder agregar un hobbie
  agregarHobbie(item: string) {
    this.hobbies.push(item);
  }

  vaciarHobbies() {
    this.hobbies = [];
  }

  getNombre(): string {
    return this.nombre;
  }

  getListaHobbies() {
    return this.hobbies;
  }

  setNombre(nombre: string) {
    this.nombre = nombre;
  }

  setApellido(apellido: string) {
    this.apellido = apellido;
  }

  setEdad(edad: number) {
    this.edad = edad;
  }
}
